source/slang/slang-emit-c-like.cpp

```diff
@@ -2790,17 +2838,59 @@ void CLikeSourceEmitter::emitClass(IRClassType* classType)
     {
         return;
     }
-
+    List<IRWitnessTable*> comWitnessTables;
+    for (auto child : classType->getDecorations())
+    {
+        if (auto decoration = as<IRCOMWitnessDecoration>(child))
+        {
+            comWitnessTables.add(cast<IRWitnessTable>(decoration->getWitnessTable()));
+        }
+    }
     m_writer->emit("class ");
 
     emitPostKeywordTypeAttributes(classType);
 
     m_writer->emit(getName(classType));
-    m_writer->emit(" : public RefObject");
+    if (comWitnessTables.getCount() == 0)
+    {
+        m_writer->emit(" : public RefObject");
+    }
+    else
+    {
+        m_writer->emit(" : public ComObject");
+        for (auto wt : comWitnessTables)
+        {
+            m_writer->emit(", public ");
+            m_writer->emit(getName(wt->getConformanceType()));
+        }
+    }
     m_writer->emit("\n{\n");
     m_writer->emit("public:\n");
     m_writer->indent();
 
+    if (comWitnessTables.getCount())
+    {
+        m_writer->emit("SLANG_COM_OBJECT_IUNKNOWN_ALL\n");
+        m_writer->emit("void* getInterface(const Guid & uuid)\n{\n");
+        m_writer->indent();
+        m_writer->emit("if (uuid == ISlangUnknown::getTypeGuid()) return static_cast<ISlangUnknown*>(this);\n");
+        for (auto wt : comWitnessTables)
```

Comment:
Does this include derived types interfaces?

Reply:
This PR does not handle derived interfaces specifically. The intention is to leave that into a separate front-end pass that flattens the inheritance list by transitively including all derived types, so any future passes will not have to worry about this.